parser/walker.go

```diff
@@ -8,40 +8,38 @@ import (
 )
 
 type walkingParser struct {
-	parser   core.Parser
-	language core.Language
-	walker   core.SourceWalker
+	parser core.Parser
+	walker core.SourceWalker
 }
 
 var _ core.TreeWalker = (*walkingParser)(nil)
 
-func NewWalkingParser(walker core.SourceWalker, language core.Language) (*walkingParser, error) {
-	parser, err := NewParser(language)
+func NewWalkingParser(walker core.SourceWalker) (*walkingParser, error) {
```

Comment:
If we are going with pool of parsers in `parserWrapper` then take the array of Lang as input for allowed langs and pass the same to parser.